app/uk/gov/hmrc/cdsimportsddsfrontend/services/CustomsDeclarationsService.scala

```diff
@@ -40,18 +40,24 @@ class CustomsDeclarationsService @Inject()(appConfig: AppConfig, declarationXml:
   def submit(eori: Eori, declarationViewModel: DeclarationViewModel)(implicit hc: HeaderCarrier): Future[DeclarationServiceResponse] = {
     val declaration: Declaration = declarationViewModel.toDeclaration
     val xml = declarationXml.fromImportDeclaration(declaration)
-    submit(eori, xml)
+    val lrn = declarationViewModel.documentationAndReferences.localReferenceNumber
+    submit(eori, xml, lrn)
   }
 
-  def submit(eori: Eori, xml: Elem)(implicit hc: HeaderCarrier): Future[DeclarationServiceResponse] = {
+  def submit(eori: Eori, xml: Elem, lrn: Option[String])(implicit hc: HeaderCarrier): Future[DeclarationServiceResponse] = {
     httpClient.POSTString[CustomsDeclarationsResponse](
       appConfig.declarationsApi.submitEndpoint, xml.toString(), headers = headers(eori))(responseReader, hc, executionContext)
       .map { customsDeclarationsResponse: CustomsDeclarationsResponse =>
         log.info("Response from Declaration API: " + customsDeclarationsResponse);
+        lrn.map(number => saveDeclaration(eori, number))
         DeclarationServiceResponse(DeclarationXml.prettyPrintToHtml(xml), customsDeclarationsResponse.status, customsDeclarationsResponse.conversationId)
       }
   }
 
+  def saveDeclaration(eori: Eori, lrn: String)(implicit hc: HeaderCarrier): Future[HttpResponse] = {
```

Comment:
This should be moved out of the service and into a connector. Have a look at Exports front end as I think they follow that pattern

Reply:
as far as I know, all the HMRC services follow that pattern but given we didn't follow that pattern yet, I didn't go for it

Reply:
agreed, i think we'd like to move towards the connector pattern